src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/letters/InvalidCharacterCheck.java

```diff
@@ -62,4 +61,46 @@ private static List<String> findAllParties(CaseData caseData) {
         }
         return parties;
     }
+
+    private static List<String> checkNamesForInvalidCharactersListingType(ListingType listingType) {
+        List<String> nameOfParties = findAllListingTypeParties(listingType);
+        return addInvalidCharsErrors(nameOfParties, listingType.getElmoCaseReference(), "cause list");
+    }
+
+    private static List<String> findAllListingTypeParties(ListingType listingType) {
+        List<String> parties = new ArrayList<>();
+        if (!Strings.isNullOrEmpty(listingType.getRespondent())) {
+            parties.add("Respondent " + listingType.getRespondent());
+        }
+        if (!Strings.isNullOrEmpty(listingType.getClaimantName())) {
+            parties.add("Claimant " + listingType.getClaimantName());
+        }
+        if (!Strings.isNullOrEmpty(listingType.getRespondentRepresentative())) {
+            parties.add("Respondent Rep " + listingType.getRespondentRepresentative());
+        }
+        if (!Strings.isNullOrEmpty(listingType.getClaimantRepresentative())) {
+            parties.add("Claimant Rep " + listingType.getClaimantRepresentative());
+        }
+        return parties;
+    }
+
+    public static boolean invalidCharactersExistAllListingTypes(ListingDetails listingDetails, List<String> errors) {
+        List<ListingTypeItem> listingTypeItems = listingDetails.getCaseData().getListingCollection();
+        List<String> invalidCharErrors = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(listingTypeItems)) {
+            for (ListingTypeItem listingTypeItem : listingTypeItems) {
+                ListingType listingType = listingTypeItem.getValue();
+                invalidCharErrors.addAll(InvalidCharacterCheck.checkNamesForInvalidCharactersListingType(listingType));
```

Comment:
no need to reference invalidCharacterCheck as same class

Reply:
removed